crowbar_framework/app/models/node_object.rb

```diff
@@ -1072,6 +1072,15 @@ def file_exist?(file)
     out[:exit_code].zero?
   end
 
+  def script_finished(script)
+    base = "/var/lib/crowbar/upgrade/" + File.basename(script, ".sh")
+    ok_file = base + "-ok"
+    failed_file = base + "-failed"
+    return "ok" if file_exist? ok_file
+    return "failed" if file_exist? failed_file
```

Comment:
Well... I don't know. I do not see it as a problem.
And I see it as more readable than the alternative threads option.

> And we call script_finished every two seconds on every compute node

Only for those where we do not have the result yet.